conans/client/cmd/profile.py

```diff
@@ -21,13 +21,23 @@ def _get_profile_keys(key):
 
 
 def cmd_profile_list(cache_profiles_path, output):
-    folder = cache_profiles_path
-    if os.path.exists(folder):
-        return [name for name in os.listdir(folder)
-                if not os.path.isdir(os.path.join(folder, name))]
-    else:
+    profiles = []
+    if os.path.exists(cache_profiles_path):
+        for current_directory, dirs, files in os.walk(cache_profiles_path):
+            # We don't want to list files from hidden directory
+            dirs[:] = [d for d in dirs if not d.startswith('.')]
+
+            for filename in files:
+                relative_to_profile_dir = os.path.relpath(current_directory, cache_profiles_path)
+                if relative_to_profile_dir == '.':
+                    profiles.append(filename)
+                else:
+                    profiles.append(relative_to_profile_dir + os.sep + filename)
+
+    if len(profiles) == 0:
         output.info("No profiles defined")
-        return []
+    profiles.sort()
```

Comment:
No need to sort here, already sorted by the OS in the `os.walk` call

Reply:
Are you sure about this? The doc says `os.walk()` uses `scandir()` or `listdir()` (depending on Python version) and apparently `The entries are yielded in arbitrary order` (https://docs.python.org/3/library/os.html#os.scandir).

Reply:
Ok, leave it, for sure it won't be a performance issue... I expect no one to have so many profiles.